clusterloader2/pkg/test/simple_test_executor.go

```diff
@@ -60,13 +60,20 @@ func (ste *simpleTestExecutor) ExecuteTest(ctx Context, conf *api.Config) *error
 	if err := ctx.GetChaosMonkey().Init(conf.ChaosMonkey, stopCh); err != nil {
 		return errors.NewErrorList(fmt.Errorf("error while creating chaos monkey: %v", err))
 	}
-	automanagedNamespacesList, err := ctx.GetClusterFramework().ListAutomanagedNamespaces()
+	automanagedNamespacesList, staleAutomanagedNamespacesList, err := ctx.GetClusterFramework().ListAutomanagedNamespaces()
 	if err != nil {
 		return errors.NewErrorList(fmt.Errorf("automanaged namespaces listing failed: %v", err))
 	}
 	if len(automanagedNamespacesList) > 0 {
 		return errors.NewErrorList(fmt.Errorf("pre-existing automanaged namespaces found"))
 	}
+	if len(staleAutomanagedNamespacesList) > 0 {
+		klog.Warningf("stale automanaged namespaces found")
+		if errList := ctx.GetClusterFramework().DeleteStaleAutomanagedNamespaces(staleAutomanagedNamespacesList); !errList.IsEmpty() {
+			klog.Warningf("stale automanaged namespace cleanup error: %v", errList.String())
+		}
+	}
+
```

Comment:
Guard this logic behind a flag. I can imagine a situation, when we don't want clusterloader to delete any namespaces.